### PR TITLE
fix(macOS): change app icon to be more inline with system design

### PR DESCRIPTION
**Pull request by zbaylin**
@bryphe and I were talking about how the current Oni2 logo on macOS doesn't really fit the current design language of the OS, specifically with the filled in background.

This PR adds a couple things:
- an SVG of the raw Oni2 icon, which can be used to generate any resolution bitmap
- an SVG of the Oni2 icon with an outer glow to be used as the icon on macOS
- an ICNS of the aforementioned macOS SVG
  - this was generated from the SVG using [Image2Icon](http://www.img2icnsapp.com/), which generates all the required resolutions in the macOS spec for icons

This is purely just a shot at what I think improves the look a little bit, but I am 100% open to feedback/input. I'm certainly no graphic designer!

Here's what the new icon looks like:
Onivim2

The outer glow color is the same as the dark-blue/violet background in the full logo.

Here's what it looks like in Finder/other contexts:
Dark mode:
Screen Shot 2020-05-23 at 11 38 56 AM
Light mode:
Screen Shot 2020-05-23 at 11 38 39 AM

For comparison, here's how the old icon looks relative to other macOS apps:
image

Please don't hold back your critiques, haha!

### CODE REVIEW

**Comment by bryphe** [comment]
Looks great, @zbaylin ! Thank you so much for improving this.

I know one piece of UI we had particular issues with was in the `cmd`+`tab` UI: https://github.com/onivim/oni2/issues/1166

Does this fix that piece of UI, too? I think I'm still seeing the old icon, but not sure if it's maybe picking up a different version of the app.

**Comment by zbaylin** [comment]
That part is set by SDL2. I can bump up the resolution of the icon.png using that SVG and see if that fixes it!

**Comment by bryphe** [comment]
Oh, interesting - didn't realize that! I guess that `SDL_SetWindowIcon` method?

>  I can bump up the resolution of the icon.png using that SVG and see if that fixes it!

Sounds great!

**Comment by zbaylin** [comment]
@bryphe I actually came up with a better idea. If SDL doesn't set the app icon, it just uses the one of the bundle (`Onivim2.icns`). This is definitely preferable because ICNS files have multiple correctly scaled versions of the icon for each resolution. So I added a flag to `Info.plist` to set the environment variable `ONI2_BUNDLED` to `1` when running in .app mode. In `Oni2_editor`, if this value is 1 it doesn't set the icon through SDL. Kinda hard to show in a screenshot, but it's pretty obvious when you check it out yourself!

File: src/bin_editor/Oni2_editor.re [hunk 1]
```diff
@@ -120,14 +120,28 @@ let createWindow = (~forceScaleFactor, ~workingDirectory, app) => {
     | _ => true
     };
 
+  let icon =
+    switch (Revery.Environment.os) {
+    | Mac =>
+      switch (Sys.getenv_opt("ONI2_BUNDLED")) {
```
Comment (bryphe) [inline src/bin_editor/Oni2_editor.re:126]: Cool idea! I didn't realize you could set environment variables via the `plist`

File: src/bin_editor/Oni2_editor.re [hunk 2]
```diff
@@ -120,14 +120,28 @@ let createWindow = (~forceScaleFactor, ~workingDirectory, app) => {
...
+      | Some(bundled) =>
+        switch (bundled) {
+        | "1" => None
+        | _ => Some("logo.png")
+        }
```
Comment (bryphe) [inline src/bin_editor/Oni2_editor.re:131]: I think we could simplify this - `Some(_) => None` (whenever the `ONI2_BUNDLED` env variable is set at all, just use `None`)

File: scripts/release.js [hunk 3]
```diff
@@ -133,7 +133,10 @@ if (process.platform == "linux") {
                 CFBundleTypeRole: fileAssoc.role,
                 CFBundleTypeIconFile: "macDocumentIcons/" + fileAssoc.icon.mac
             }
-        })
+        }),
+      LSEnvironment: {
```
Comment (bryphe) [inline scripts/release.js:137]: Neat! Didn't know this was an option

**Comment by bryphe** [comment]
> @bryphe I actually came up with a better idea. If SDL doesn't set the app icon, it just uses the one of the bundle (Onivim2.icns). This is definitely preferable because ICNS files have multiple correctly scaled versions of the icon for each resolution.

That makes sense - great idea 👍 This will be better at handling things like retina displays for sure.

**Comment by bryphe** [comment]
Looks so much better! Fixes #1166 👍

File: src/bin_editor/Oni2_editor.re (hunk 2 above)
Comment (zbaylin) [reply src/bin_editor/Oni2_editor.re]: Awesome I just changed that in the most recent commit. Thanks!

**Pull request merged**